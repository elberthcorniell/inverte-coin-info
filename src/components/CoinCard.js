import PropTypes from 'prop-types';
import { formatNumber } from '../utils';

const CoinCard = ({
  name, image, symbol, index,
  currentPrice, marketCap,
}) => (
  <div
    className="gray-card"
    style={{
      position: 'relative',
      textAlign: 'center',
    }}
  >
    <strong style={{
      position: 'absolute',
      top: 10,
      left: 10,
    }}
    >
      {index + 1}
    </strong>
    <img
      src={image}
      alt={name}
      height={40}
      style={{
        border: 'none',
      }}
    />
    <div style={{
      margin: '10px 0',
    }}
    >
      <strong>{name}</strong>
      {' '}
      <strong><span>{symbol}</span></strong>
    </div>
    <ul style={{
      textAlign: 'left',
      marginTop: 40,
    }}
    >
      <li>
        <strong style={{
          fontWeight: 300,
        }}
        >
          Price:
        </strong>
        <span>
          $
          {formatNumber(currentPrice, 2)}
        </span>
      </li>
      <li>
        <strong style={{
          fontWeight: 300,
        }}
        >
          Market Cap:
        </strong>
        <span>
          $
          {formatNumber(marketCap, 2)}
        </span>
      </li>
    </ul>
  </div>
);

CoinCard.propTypes = {
  name: PropTypes.string,
  image: PropTypes.string,
  symbol: PropTypes.string,
  index: PropTypes.number,
  currentPrice: PropTypes.number,
  marketCap: PropTypes.number,
};

CoinCard.defaultProps = {
  name: '',
  image: '',
  symbol: '',
  index: 0,
  currentPrice: 0,
  marketCap: 0,
};

export default CoinCard;
