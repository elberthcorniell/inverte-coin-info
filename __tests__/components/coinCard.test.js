import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CoinCard from '../../src/components/CoinCard';

describe('test', () => {
  test('Render as spected', () => {
    render(<CoinCard
      index={1}
      name="Bitcoin"
      symbol="BTC"
      image="./"
      currentPrice={30_000}
      marketCap={600_000_000_000}
    />);
    expect(document.body.innerHTML).toMatchSnapshot();
  });

  test('Loads and displays market cap correctly', () => {
    render(<CoinCard
      index={1}
      name="Bitcoin"
      symbol="BTC"
      image="./"
      currentPrice={30_000}
      marketCap={600_000_000_000}
    />);
    const text = screen.getAllByText('$600,000,000,000.00');
    expect(typeof text[0]).not.toBe('undefined');
  });

  test('Loads and displays current price correctly', () => {
    render(<CoinCard
      index={1}
      name="Bitcoin"
      symbol="BTC"
      image="./"
      currentPrice={30_000}
      marketCap={600_000_000_000}
    />);
    const text = screen.getAllByText('$30,000.00');
    expect(typeof text[0]).not.toBe('undefined');
  });
});
