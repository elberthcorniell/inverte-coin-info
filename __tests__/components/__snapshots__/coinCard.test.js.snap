// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Render as spected 1`] = `"<div><div class=\\"gray-card\\" style=\\"position: relative; text-align: center;\\"><strong style=\\"position: absolute; top: 10px; left: 10px;\\">2</strong><img src=\\"./\\" alt=\\"Bitcoin\\" height=\\"40\\"><div style=\\"margin: 10px 0px;\\"><strong>Bitcoin</strong> <strong><span>BTC</span></strong></div><ul style=\\"text-align: left; margin-top: 40px;\\"><li><strong style=\\"font-weight: 300;\\">Price:</strong><span>$30,000.00</span></li><li><strong style=\\"font-weight: 300;\\">Market Cap:</strong><span>$600,000,000,000.00</span></li></ul></div></div>"`;
